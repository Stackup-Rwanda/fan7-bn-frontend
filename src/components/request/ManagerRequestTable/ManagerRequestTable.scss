.requests-table {
  width: 100%;
  padding-top: 1em;
  background: white;

  &__header {
    border-top: 0.5px solid rgba(211, 210, 210, 0.5);
    border-bottom: 0.5px solid rgba(211, 210, 210, 0.5);

    label {
      display: flex;
      flex-direction: row;
      justify-content: flex-end;
      align-items: center;
      padding: 0.5em 1em;
      font-size: 14px;
      font-family: 'Poppins', sans-serif;
    }

    span {
      color: #555252 !important;
      margin-right: 1em;
    }

    &___search {
      display: block;
      width: auto !important;
      height: 14px !important;
      padding: 6px 12px;
      box-sizing: border-box;
      color: #555 !important;
      background-color: #ffffff;
      background-image: none;
      border: 1px solid rgba(0, 0, 0, 0.4) !important;
      border-radius: 0 !important;
      margin-bottom: 0;

      &:focus {
        outline: 0;
        -webkit-box-shadow: inset 0 0.5px 0.5px rgba(0, 0, 0, 0.075), 0 0 8px rgba(5, 108, 242, 0.6);
        box-shadow: inset 0 0.5px 0.5px rgba(0, 0, 0, 0.075), 0 0 8px rgba(5, 108, 242, 0.6);
      }

      &::-moz-placeholder {
        color: #999;
        opacity: 1;
      }

      &:-ms-input-placeholder {
        color: #999;
      }

      &::-webkit-input-placeholder {
        color: #999;
      }
    }
  }

  table {
    width: 100%;
    text-align: center;
    font-size: 14px;
    padding: 0 .5em;
    border-bottom: 0.5px solid rgba(211, 210, 210, 0.5);

    th,
    td {
      padding: 10px 5px;
    }

    th {
      border-bottom: 0.2px solid rgba(211, 210, 210, 0.5);
    }

    tr {
      &:nth-child(even) {
        background-color: rgba(236, 236, 236, 0.5);
      }
    }

    td {
      button {
        display: block;
        font-weight: normal;
        text-align: center;
        white-space: nowrap;
        background-color: transparent;
        background-image: none;
        -ms-touch-action: manipulation;
        touch-action: manipulation;
        -webkit-user-select: none;
        -moz-user-select: none;
        -ms-user-select: none;
        user-select: none;
        border: none;
        cursor: pointer;

        &:focus,
        &:active &:active:focus {
          outline: 0;
          background-image: none;
        }
      }

      img {
        height: 1.5em;
      }
    }
  }
}
