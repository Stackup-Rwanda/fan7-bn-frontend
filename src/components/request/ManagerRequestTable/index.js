import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import Spinner from '../../Spinner';
import './ManagerRequestTable.scss';
import options from '../../../assets/icons/icons8-menu-vertical-30.png';

const Table = ({ requests, handleSearch, loading, handleAction }) => {
  return (
    <div className="requests-table">
      <div className="requests-table__header">
        <label htmlFor="search">
          <span>Search:</span>
          <input
            type="text"
            className="requests-table__header___search"
            name="search"
            placeholder="Search..."
            onChange={handleSearch}
          />
        </label>
      </div>
      <table className="table">
        <thead>
          <tr>
            <th>Requester</th>
            <th>Trip Reason</th>
            <th>Destination(s)</th>
            <th>Travel Date</th>
            <th>Return Date</th>
            <th>Type</th>
            <th>Status</th>
            <th>Action</th>
          </tr>
        </thead>
        <tbody>
          {!!loading ? (<Spinner />) : (!!requests ? (
            requests.map(request => (
              <tr key={request.id}>
                <td>{request.user.email}</td>
                <td>{request.reason}</td>
                <td>
                  {request.destination && request.destination.map(destination => `${destination}  `)}
                </td>
                <td>
                  {request.travel_date &&
                    request.travel_date.map(travelDate => `${moment(travelDate).format('L')}  `)}
                </td>
                <td>{request.return_date ? moment(request.return_date).format('L') : ''}</td>
                <td>{request.type}</td>
                <td>{request.status}</td>
                <td>
                  <button type="button" onClick={handleAction}>
                    <img src={options} alt="more options" data-requestId={request.id} />
                  </button>
                </td>
              </tr>
            ))
          ) : (
            <tr>
              <td colSpan="8">No trip requests found</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

Table.propTypes = {
  requets: PropTypes.arrayOf(),
  handleSearch: PropTypes.func.isRequired,
  loading: PropTypes.bool,
};

Table.defaultProps = {
  requets: [],
  loading: false,
};

export default Table;
