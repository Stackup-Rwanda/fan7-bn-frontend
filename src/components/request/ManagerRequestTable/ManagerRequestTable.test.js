/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount, shallow } from 'enzyme';
import ManagerRequestTable from './';

describe('ManagerRequestTable Component Test', () => {
  it('should return true if component exists', () => {
    const wrapper = shallow(<ManagerRequestTable />);

    expect(wrapper.find('table').exists()).toBe(true);
  });

  it('should render appropriate props', () => {
    const props = {
      requests: [
        {
          id: 11,
          accommodation_id: 1,
          origin: 'rwanda, kigali',
          destination: ['burundi, bujumbura'],
          travel_date: ['2020-06-24'],
          return_date: null,
          reason: 'dddddddddd',
          status: 'Pending',
          user: { email: 'elvis@gmail.com' },
          accommodation: { name: 'Tam Tam' },
        },
        {
          id: 10,
          accommodation_id: null,
          origin: 'rwanda, kigali',
          destination: ['burundi, bujumbura'],
          travel_date: ['2020-05-22'],
          return_date: null,
          reason: 'dddddddddd',
          status: 'Pending',
          user: { email: 'elvis@gmail.com' },
          accommodation: { name: 'Tam Tam' },
        },
      ],
    };
    const wrapper = shallow(<ManagerRequestTable {...props} />);

    expect(wrapper.find('tbody tr').exists()).toBe(true);
  });
});

describe('<ManagerRequestTable /> shallow rendering tests', () => {
  it('matches the snapshot', () => {
    const props = {
      requests: [
        {
          id: 11,
          accommodation_id: 1,
          origin: 'rwanda, kigali',
          destination: ['burundi, bujumbura'],
          travel_date: ['2020-06-24'],
          return_date: null,
          reason: 'dddddddddd',
          status: 'Pending',
          user: { email: 'elvis@gmail.com' },
          accommodation: { name: 'Tam Tam' },
        },
        {
          id: 10,
          accommodation_id: 1,
          origin: 'rwanda, kigali',
          destination: ['burundi, bujumbura'],
          travel_date: ['2020-05-22'],
          return_date: null,
          reason: 'dddddddddd',
          status: 'Pending',
          user: { email: 'elvis@gmail.com' },
          accommodation: { name: 'Tam Tam' },
        },
      ],
    };
    const tree = shallow(<ManagerRequestTable {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
