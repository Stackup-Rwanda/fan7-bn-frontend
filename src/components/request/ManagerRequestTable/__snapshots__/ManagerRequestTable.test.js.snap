// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManagerRequestTable /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="requests-table"
>
  <div
    className="requests-table__header"
  >
    <label
      htmlFor="search"
    >
      <span>
        Search:
      </span>
      <input
        className="requests-table__header___search"
        name="search"
        placeholder="Search..."
        type="text"
      />
    </label>
  </div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Requester
        </th>
        <th>
          Trip Reason
        </th>
        <th>
          Destination(s)
        </th>
        <th>
          Travel Date
        </th>
        <th>
          Return Date
        </th>
        <th>
          Type
        </th>
        <th>
          Status
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="11"
      >
        <td>
          elvis@gmail.com
        </td>
        <td>
          dddddddddd
        </td>
        <td>
          burundi, bujumbura  
        </td>
        <td>
          06/24/2020  
        </td>
        <td />
        <td />
        <td>
          Pending
        </td>
        <td>
          <button
            type="button"
          >
            <img
              alt="more options"
              data-requestId={11}
              src={
                Object {
                  "process": [Function],
                }
              }
            />
          </button>
        </td>
      </tr>
      <tr
        key="10"
      >
        <td>
          elvis@gmail.com
        </td>
        <td>
          dddddddddd
        </td>
        <td>
          burundi, bujumbura  
        </td>
        <td>
          05/22/2020  
        </td>
        <td />
        <td />
        <td>
          Pending
        </td>
        <td>
          <button
            type="button"
          >
            <img
              alt="more options"
              data-requestId={10}
              src={
                Object {
                  "process": [Function],
                }
              }
            />
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
